cg/clients/freshdesk/freshdesk_client.py

```diff
@@ -11,28 +18,51 @@
 class FreshdeskClient:
     """Client for communicating with the freshdesk REST API."""
 
-    def __init__(self):
-        self.headers = None
-        self.api_key = None
-        self.url = None
-
-    def init_app(self, url: str, api_key: str):
-        """Set up the client."""
-        self.url = url
+    def __init__(self, base_url: str, api_key: str):
+        self.base_url = base_url
         self.api_key = api_key
-        self.headers = {"Content-Type": "application/json"}
+        self.session = self._configure_session()
+
+    def __enter__(self) -> "FreshdeskClient":
+        return self
 
-    @property
-    def auth_header(self):
-        return self.api_key, "X"
+    def __exit__(self, exc_type, exc_value, traceback) -> None:
+        self.session.close()
 
     def create_ticket(self, ticket: TicketCreate) -> TicketResponse:
         """Create a ticket."""
         LOG.debug(ticket.model_dump_json())
-        response = requests.post(
-            url=f"{self.url}{EndPoints.TICKETS}",
-            headers=self.headers,
-            auth=self.auth_header,
-            json=ticket.model_dump(exclude_none=True),
+        try:
+            response: Response = self.session.post(
+                url=f"{self.base_url}{EndPoints.TICKETS}",
+                json=ticket.model_dump(exclude_none=True),
+            )
+            response.raise_for_status()
+            ticket_response: TicketResponse = TicketResponse.model_validate(response.json())
+        except RequestException as error:
+            LOG.error(f"Could not create ticket: {error}")
+            raise FreshdeskAPIException(error) from error
+        except ValidationError as error:
+            LOG.error(f"Response from Freshdesk does not fit model: {TicketResponse}.\n{error}")
+            raise FreshdeskModelException(error) from error
+        return ticket_response
+
+    def _configure_session(self) -> Session:
+        """Configures and sets a session to be used for requests."""
+        session = Session()
+        self._configure_retries(session)
+        session.auth = (self.api_key, "X")
+        session.headers.update({"Content-Type": "application/json"})
+        return session
+
+    @staticmethod
+    def _configure_retries(session: Session) -> None:
+        """Configures retries for the session."""
+        retry_strategy = Retry(
+            total=5,
+            status_forcelist=[TO_MANY_REQUESTS_CODE],
+            allowed_methods=None,
+            backoff_factor=2,
         )
```

Comment:
This strategy means:
1. Retry 5 times
2. On every retry wait twice as long (2,4,8,16,32) seconds.
3. This should only be done on 429 (too many requests) exit status
4. No restrictions on which methods this applies to

Reply:
By default retries is only applied to idempotent methods (ie not POST) but we do want it for POST as well. 

That's why I only added 429 for the strategy, but please give your opinion on how this should be configured

Reply:
A bit unsure of what this means. What does "this" in point 3 refer to? The retries? Because if we only send one request and we get an error, that would not be a "too many requests" error?

Reply:
Yes, "this" refers to retries. Basically if we get any other error exit code than 429 we will not retry.